### ISSUE TITLE
Configuration issue / Token renewal operation failed due to timeout

### ISSUE THREAD

**Issue by tmschbgr** [matt-ankerson/vue-adal#1]
Hey mate,

First of all, thank you so much for this example, this is gold.

I think I am really close to make it working, but I am struggling with finding the right configuration. Currently I am getting an Token renewal operation failed due to timeout issue if I call the acquireToken() function.

In your ./authentication/index.js file you have tenant, clientId, redirectUri and cacheLocation. I try to figure out what those settings are, but unfortunately the Microsoft documentation is not very consistent with their naming.

  tenant: 'xxxxxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  clientId: 'xxxxxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  redirectUri: 'http://localhost:8080',
  cacheLocation: 'localStorage'

**Tenant**: Is that right that the tenant is an Id? Found as Directory Id in Azure Portal under AAD > Properties
**ClientId** is clear for me
**redirectUri**: Is that just the base url of your app? or would it be something like http://localhost:8080/auth/redirect?

Also I noticed that in the same file you still have 2 placeholder for the functions acquireToken and acquireTokenRedirect "azure active directory resource id"

**Active Directory Resource Id**: So is that the instance? like 'http://login.microsoftonline.com'?

Thank you very much for sharing your work so far, you are a legend

Cheers

**Comment by matt-ankerson** [comment]
Hey, I'm glad this example is useful  :). Please let me know if anything remains unclear.

Yes, **Tenant** is your Directory Id found in the Azure Portal under the AAD properties. Or you could use the domain name of your directory - i.e. `MySampleAAD.onmicrosoft.com`.
**ClientId** is sometimes called *ApplicationId*, and represents the registered application in AAD.
In this case, **redirectUri** is indeed the base url of the app - so probably `http://localhost:8080`. Your `redirectUri` needs to be included as a `replyUrl` in your registered AAD application.

When you're acquiring a token for a resource, you need to supply a **Resource Identifier** belonging to another registered application in AAD which represents the resource you're requesting. In my case, the identifier is the base url for my protected api - `https://my-secure-api.net`. This resource identifier should be present in the `identifierUris` collection in the registered AAD app representing your protected resource. I think it's also possible to use the ClientId of your protected resource - but I can't remember.

Additionally, you'll need to grant permission for your application to access the other protected application. You can do this in the Azure Portal.

So, the config for your locally hosted SPA could look something like this:

``` Javascript
const config = {
  tenant: '6b87e902-55c6-450c-ada6-63c04e5a7d5c',
  clientId: '2461d95b-4a91-4dc9-8fcc-ab46077d70cf',
  redirectUri: 'http://localhost:8080',
  cacheLocation: 'localStorage'
};
const protectedResourceIdentifier = 'https://my-secure-api.net';
```

**Comment by tmschbgr** [comment]
Hi Matt, thank you very much for the explanation, that helps a lot.

So in my case the SPA and the API are sitting in the same Azure App Service and therefore are having the same AAD App Id. I figured out that I have to use the clientId again as protectedResourceIdentifier, then it works. When I tried with 'https://mywebsite.azurewebsites.net' I was getting the following error:
`AADSTS90009: Application '21ae8ccc-a45d-4e6f-8c74-7bfd0f2da178' is requesting a token for itself. This scenario is supported only if resource is specified using the GUID based App Identifier.
Trace ID: 750a798b-35c3-4e8f-9516-4313a1e13500
Correlation ID: ab8c9468-7566-40e3-b1a2-4ef2a2b47853
Timestamp: 2018-03-15 20:20:22Z`

Thank you for your support. I think that is a really easy approach on how to authenticate users in an Azure environment. 

I just released that you are from NZ as well, small world ;-)

Cheers
